### PR TITLE
sbot: init at 9d686ee70c6b2073da626fa08e38a8e988b5c5b4

### PR DESCRIPTION
**Pull request by matthiasbeyer**
###### Motivation for this change

Package `sbot`

###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

I need help here. That's actually the first application from the node-js ecosystem I'm trying to package.

### CODE REVIEW

File: pkgs/applications/networking/sbot/default.nix [hunk 1]
```diff
@@ -0,0 +1,738 @@
...
+
+let
+	np = nodePackages.override {
+		generated = ./generated.nix;
+	};
+in
+
+np.buildNodePackage rec {
+	version = "9.4.2";
+  name = "sbot-${version}";
```
Comment (grahamc) [inline pkgs/applications/networking/sbot/default.nix]: Couple things,

1. you're actually doing `init at 9.4.2` -- can you update the commit message?
2. can you dedent `version` back to the normal level here?

**Comment by matthiasbeyer** [comment]
@grahamc fixed what you mentioned, it still does not build though, and I do not know how to resolve this error:

```
error: anonymous function at /home/m/archive/development/contrib/nixpkgs/pkgs/development/node-packages/default-v6.nix:1:1 called with unexpected argument ‘generated’, at /home/m/archive/development/contrib/nixpkgs/lib/customisation.nix:56:12
```

**Comment by fpletz** [comment]
We should not include a 12k lines file just for one package that probably nobody uses anyway.

Can't this be merged with `./pkgs/top-level/node-packages-generated.nix` somehow? I don't know/care about nodejs.

**Review by fpletz — CHANGES_REQUESTED** [review]
Please merge the generated dependes with `./pkgs/top-level/node-packages-generated.nix` if possible.

Adding more than 12k lines for a single package that isn't very popular is not acceptable.

**Comment by matthiasbeyer** [comment]
@fpletz I would love to - is there a tool available for this or do I have to do it by hand?

**Comment by Mic92** [comment]
@matthiasbeyer add it to pkgs/development/node-packages/node-packages.json and run `generate.sh` in the same directory.

**Comment by matthiasbeyer** [comment]
@Mic92 umm... I'm not sure what you mean? Add the .nix file to the .json file? Sorry if I'm dumb...

**Comment by Mic92** [comment]
Add your package to `node-packages.json` and then regenerate node-packages-v4.nix and node-packages-v6.nix with the script `generate.sh`

**Comment by matthiasbeyer** [comment]
@Mic92 Hope that is what you suggested...

**Pull request closed without merging**